core/src/main/java/edu/wpi/grip/core/http/HttpPipelineSwitcher.java

```diff
@@ -0,0 +1,60 @@
+package edu.wpi.grip.core.http;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import edu.wpi.grip.core.serialization.Project;
+import edu.wpi.grip.core.util.GRIPMode;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jetty.server.Request;
+
+import static edu.wpi.grip.core.util.GRIPMode.HEADLESS;
+
+/**
+ * Jetty handler responsible for loading pipelines sent over HTTP.
+ */
+@Singleton
+public class HttpPipelineSwitcher extends PedanticHandler {
+
+    private final Project project;
+    private final GRIPMode mode;
+
+    @Inject
+    HttpPipelineSwitcher(Project project, GRIPMode mode) {
+        super(GripServer.PIPELINE_UPLOAD_PATH, true);
+        this.project = project;
+        this.mode = mode;
+    }
+
+    @Override
+    protected void handleIfPassed(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        if (!isPost(request)) {
+            response.setStatus(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+            baseRequest.setHandled(true);
+            return;
+        }
+        switch (mode) {
+            case HEADLESS:
+                project.open(new String(IOUtils.toByteArray(request.getInputStream()), "UTF-8"));
```

Comment:
Is it worth replacing "UTF-8" with `StandardCharsets.UTF_8.name()`?

[StandardCharsets (Java Platform SE 8)](https://docs.oracle.com/javase/8/docs/api/java/nio/charset/StandardCharsets.html)